Add missing Knowledge Checks, Learning Outcomes, and Knowledge Check links to Ruby on Rails/A Railsy Web Refresher (#23008)
TheOdinProject/curriculum: rails_programming/rails_basics/web_refresher.md

To really understand how Rails works, you need to have a solid base in the guts of the web.  You've already covered some of this in previous sections (and if you went through the Ruby curriculum you can just skim it), but this time you'll get a chance (in the project) to reach out and make some real web requests.

### Learning Outcomes
By the end of this lesson, you should be able to do the following:

> * Deconstruct an HTTP-message.
> * Name the 4 most common verbs used in HTTP-requests.
> * Describe the 7 RESTful actions we differentiate in Rails.
> * Recognize all components of a URL.
> * Identify the different components of an MVC-pattern and explain what their tasks are.
> * Describe what an API is and how it makes applications work together "seamless".
> * Explain what "cookies" and "sessions" are.
> * Know the difference between "authentication" and "authorization".

Reviewer: Per the wiki post I wrote about Learning Outcomes and Knowledge Checks we want these to be higher level, for instance here's the changes I would probably make:
```suggestion
* Explain the basics of HTTP.
* Identify the 4 most commonly used HTTP verbs.
* Identify the 7 RESTful routes of Rails.
* Name the different components of a URL.
* Explain the basics of MVC.
* Describe what an API is.
* Explain what "cookies" and "sessions" are.
* Describe the difference between "authentication" and "authorization".
```

* [HTTP explained by Harvard's David Malan](http://www.youtube.com/watch?v=8KuO4r5CHjM)
* [HTTP Request/Response Basics](http://justahelp.blogspot.com/2013/09/http-requestresponse-basics.html) from Pralay Roy

### Knowledge Check
This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.

> * <a class="knowledge-check-link" href="https://code.tutsplus.com/tutorials/http-the-protocol-every-web-developer-must-know-part-1--net-31177">How do you call an HTTP-message that goes from client to server? How do you call one that goes the opposite direction? In which header would you encounter a "status code" and in which one an "action verb"?</a>
> * <a class="knowledge-check-link" href="https://www.mattcutts.com/blog/seo-glossary-url-definitions/">How do you call the additional information that is added after the path of a URL? Which symbols precede and separate this information?</a>
> * <a class="knowledge-check-link" href="https://betterexplained.com/articles/intermediate-rails-understanding-models-views-and-controllers/">What does MVC stand for?</a>
> * <a class="knowledge-check-link" href="https://betterexplained.com/articles/intermediate-rails-understanding-models-views-and-controllers/">Which Rails-feature/script analyzes the URL to dispatch the right controller and is hence the precursor for all MVC-related actions?</a>
> * <a class="knowledge-check-link" href="https://money.howstuffworks.com/business-communications/how-to-leverage-an-api-for-conferencing1.htm">What is an "API"? Why does every right-minded website has one (or more)?</a>
> * <a class="knowledge-check-link" href="https://en.wikipedia.org/wiki/HTTP_cookie#Session_management">Why do you need "cookies" to continue your "session"?</a>
> * <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/a-railsy-web-refresher#authorization">Explain the difference between "authentication" and "authorization".</a>

Reviewer: Also per some of my suggestions in that wiki post, we want to limit Knowledge Checks to one question each. To do so I would split the first question into parts, and to keep the total number of questions down I would drop a bit from one of your MVC questions and all of another one. Here are the changes I would make (including some grammar tweaks):

```suggestion
* <a class="knowledge-check-link" href="https://code.tutsplus.com/tutorials/http-the-protocol-every-web-developer-must-know-part-1--net-31177">What do you call an HTTP message that goes from client to server?</a>
* <a class="knowledge-check-link" href="https://code.tutsplus.com/tutorials/http-the-protocol-every-web-developer-must-know-part-1--net-31177">What do you call an HTTP message that goes from server to client?</a>
* <a class="knowledge-check-link" href="https://code.tutsplus.com/tutorials/http-the-protocol-every-web-developer-must-know-part-1--net-31177">Which HTTP message would include a status code and which would include an action verb?</a>
* <a class="knowledge-check-link" href="https://www.mattcutts.com/blog/seo-glossary-url-definitions/">What is the name of the additional information that is added after the path of a URL?</a>
* <a class="knowledge-check-link" href="https://betterexplained.com/articles/intermediate-rails-understanding-models-views-and-controllers/">What does MVC stand for?</a>
* <a class="knowledge-check-link" href="https://money.howstuffworks.com/business-communications/how-to-leverage-an-api-for-conferencing1.htm">What is an "API"?</a>
* <a class="knowledge-check-link" href="https://en.wikipedia.org/wiki/HTTP_cookie#Session_management">Why do you need "cookies" to continue your "session"?</a>
* <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/a-railsy-web-refresher#authorization">What is the difference between "authentication" and "authorization"?</a>
```

* [HTTP explained by Harvard's David Malan](http://www.youtube.com/watch?v=8KuO4r5CHjM)
* [HTTP Request/Response Basics](http://justahelp.blogspot.com/2013/09/http-requestresponse-basics.html) from Pralay Roy

### Knowledge Check
This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.

* <a class="knowledge-check-link" href="https://code.tutsplus.com/tutorials/http-the-protocol-every-web-developer-must-know-part-1--net-31177">How do you call an HTTP-message that goes from client to server? How do you call one that goes the opposite direction? In which header would you encounter a "status code" and in which one an "action verb"?</a>

> * <a class="knowledge-check-link" href="https://code.tutsplus.com/tutorials/http-the-protocol-every-web-developer-must-know-part-1--net-31177">What is the difference between HTTP and HTTPS?</a>

Reviewer: I'd drop this question because it isn't very relevant and the linked resource only barely touches on it anyways.

> * <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-on-rails/lessons/a-railsy-web-refresher#rest">Name all 7 RESTful controller actions of Rails and the HTTP-verbs with which they are associated.</a>

Reviewer: I'd drop this question because Rails details will be more relevant in later lessons than this one.

* <a class="knowledge-check-link" href="https://www.mattcutts.com/blog/seo-glossary-url-definitions/">How do you call the additional information that is added after the path of a URL? Which symbols precede and seperate this information?</a>
* <a class="knowledge-check-link" href="https://betterexplained.com/articles/intermediate-rails-understanding-models-views-and-controllers/">What does MVC stand for?</a>
* <a class="knowledge-check-link" href="https://betterexplained.com/articles/intermediate-rails-understanding-models-views-and-controllers/">Which Rails-feature/script analyzes the URL to dispatch the right controller and is hence the precursor for all MVC-related actions?</a>

> * <a class="knowledge-check-link" href="https://betterexplained.com/articles/intermediate-rails-understanding-models-views-and-controllers/">Which part of MVC communicates with the database?</a>

Reviewer: I'd drop these questions because Rails details will be more relevant in later lessons than this one.

* <a class="knowledge-check-link" href="https://money.howstuffworks.com/business-communications/how-to-leverage-an-api-for-conferencing1.htm">What is an "API"? Why does every right-minded website has one (or more)?</a>
* <a class="knowledge-check-link" href="https://en.wikipedia.org/wiki/HTTP_cookie#Session_management">Why do you need "cookies" to continue your "session"?</a>

> * <a class="knowledge-check-link" href="https://www.allaboutcookies.org/">What ethical issues do cookies bring along and how should web developers anticipate those?</a>

Reviewer: I'd drop this question because like HTTPS direct relevance to learning Rails is quite low.
